mlflow/gateway/providers/openai.py

```diff
@@ -159,6 +205,52 @@ def _prepare_completion_response_payload(self, resp):
             ),
         )
 
+    async def completions_stream(
+        self, payload: completions.RequestPayload
+    ) -> AsyncIterable[completions.StreamResponsePayload]:
+        from fastapi.encoders import jsonable_encoder
+
+        payload = jsonable_encoder(payload, exclude_none=True)
+        self.check_for_model_field(payload)
+        payload = self._prepare_completion_request_payload(payload)
+
+        stream = send_stream_request(
+            headers=self._request_headers,
+            base_url=self._request_base_url,
+            path="chat/completions",
+            payload=self._add_model_to_payload_if_necessary(payload),
+        )
+
+        async for chunk in stream:
```

Comment:
Can you take a look at this issue I filed last year?

https://community.openai.com/t/incomplete-stream-chunks-for-completions-api/383520

OpenAI sometimes returns incomplete chunks. To handle this, we need a logic like this:


```python
buffer = ""
for chunk in stream:
    # Split at the last newline delimiter
    (complete, *rest) = chunk.rsplit("\n", 1)

    if rest:
        # buffer + complete should contains at least one complete chunk.
        # the number of complete chunks could be > 1
        yield parse(buffer + complete)

        # Update buffer for the next iteration
        buffer = rest[0]
    else:
        # Otherwise, we have an incomplete line
        buffer += complete
```

Reply:
i still can't reproduce this on Python, but anyway I have pushed a fix for it

Reply:
Did you use the OpenAI python SDK? I could not repro with Python either. I needed to use the JS SDK.

Reply:
Yes, I couldn't reproduce with neither python sdk or rest. It might be something with nodejs chunk splitting logic.